command/hcl2_upgrade.go

```diff
@@ -0,0 +1,294 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	texttemplate "text/template"
+
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	hcl2shim "github.com/hashicorp/packer/hcl2template/shim"
+	"github.com/hashicorp/packer/template"
+	"github.com/posener/complete"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type HCL2UpgradeCommand struct {
+	Meta
+}
+
+func (c *HCL2UpgradeCommand) Run(args []string) int {
+	ctx, cleanup := handleTermInterrupt(c.Ui)
+	defer cleanup()
+
+	cfg, ret := c.ParseArgs(args)
+	if ret != 0 {
+		return ret
+	}
+
+	return c.RunContext(ctx, cfg)
+}
+
+func (c *HCL2UpgradeCommand) ParseArgs(args []string) (*HCL2UpgradeArgs, int) {
+	var cfg HCL2UpgradeArgs
+	flags := c.Meta.FlagSet("hcl2_upgrade", FlagSetNone)
+	flags.Usage = func() { c.Ui.Say(c.Help()) }
+	cfg.AddFlagSets(flags)
+	if err := flags.Parse(args); err != nil {
+		return &cfg, 1
+	}
+	args = flags.Args()
+	if len(args) != 1 {
+		flags.Usage()
+		return &cfg, 1
+	}
+	cfg.Path = args[0]
+	if cfg.OutputFile == "" {
+		cfg.OutputFile = cfg.Path + ".pkr.hcl"
```

Comment:
It would be nice to remove the `.json` extension here. What do you think? 
Like: 
```
cfg.OutputFile = strings.TrimSuffix(cfg.Path, filepath.Ext(cfg.Path)) + ".pkr.hcl"
```

Reply:
I like that the file keep its original extension because then users get a _hint_ that it was generated from JSON.

I would also like to mark that file as 'disposable' so users use parts of it instead of working in it. 🤔  I will add an autogenerated comment at the top.

Reply:
Sounds a good reason!